src/AppInstallerCommonCore/MSStoreDownload.cpp

```diff
@@ -4,488 +4,1045 @@
+            }
+
+            std::string base64LicenseContent = JSON::GetRawStringValueFromJsonNode(keys->get().at(0), JSON::GetUtilityString(Value)).value_or("");
+            if (base64LicenseContent.empty())
+            {
+                AICLI_LOG(Core, Error, << "Missing license content");
+                THROW_HR(APPINSTALLER_CLI_ERROR_LICENSING_API_FAILED);
+            }
+
+            return JSON::Base64Decode(base64LicenseContent);
         }
-        catch (...)
+    }
+
+    namespace
+    {
+        Http::HttpClientHelper::HttpRequestHeaders GetAuthHeaders(std::unique_ptr<Authentication::Authenticator>& authenticator)
         {
-            AICLI_LOG(Core, Error, << "Received invalid information.");
+            if (!authenticator)
+            {
+                return {};
+            }
+
+            Http::HttpClientHelper::HttpRequestHeaders result;
+
+            auto authResult = authenticator->AuthenticateForToken();
+            if (FAILED(authResult.Status))
+            {
+                AICLI_LOG(Repo, Error, << "Authentication failed. Result: " << authResult.Status);
+                THROW_HR_MSG(authResult.Status, "Failed to authenticate for MicrosoftEntraId");
+            }
+            result.insert_or_assign(web::http::header_names::authorization, JSON::GetUtilityString(Authentication::CreateBearerToken(authResult.Token)));
+
+            return result;
         }
     }
 
-    std::string GetMSStoreCatalogRestApi(const std::string& productId, const std::string& locale)
+    MSStoreDownloadContext::MSStoreDownloadContext(
+        std::string productId,
+        AppInstaller::Utility::Architecture architecture,
+        Manifest::PlatformEnum platform,
+        std::string locale,
+        AppInstaller::Authentication::AuthenticationArguments authArgs) :
+        m_productId(std::move(productId)), m_architecture(architecture), m_platform(platform), m_locale(std::move(locale))
     {
-        std::string languageValue = !locale.empty() ? locale : std::string{ Neutral };
-        std::string market = AppInstaller::Runtime::GetOSRegion();
+#ifndef AICLI_DISABLE_TEST_HOOKS
+        if (!TestHooks::s_Licensing_HttpPipelineStage_Override)
+        {
+#endif
+            Authentication::MicrosoftEntraIdAuthenticationInfo licensingMicrosoftEntraIdAuthInfo;
+            licensingMicrosoftEntraIdAuthInfo.Resource = "c5e1cb0d-5d24-4b1a-b291-ec684152b2ba";
+            Authentication::AuthenticationInfo licensingAuthInfo;
+            licensingAuthInfo.Type = Authentication::AuthenticationType::MicrosoftEntraId;
+            licensingAuthInfo.MicrosoftEntraIdInfo = std::move(licensingMicrosoftEntraIdAuthInfo);
+
+            // Not moving authArgs because we'll have auth for display catalog and sfs client in the near future.
+            m_licensingAuthenticator = std::make_unique<Authentication::Authenticator>(std::move(licensingAuthInfo), authArgs);
+#ifndef AICLI_DISABLE_TEST_HOOKS

```

Comment:
style: You can leave the `{ }` in and just #def out the `if` line.  That way you don't need a second #ifdef.